implementations/rust/ockam/ockam_api/tests/authority.rs

```diff
@@ -0,0 +1,390 @@
+    attrs.insert(b"ockam-role".to_vec(), b"enroller".to_vec());
+    attrs.insert(b"trust_context_id".to_vec(), b"123456".to_vec());
+
+    for _ in 0..number_of_admins {
+        let admin = secure_channels
+            .identities()
+            .identities_creation()
+            .create_identity()
+            .await?
+            .identifier()
+            .clone();
+
+        let entry = AttributesEntry::new(attrs.clone(), now, None, None);
+        trusted_identities.insert(admin.clone(), entry);
+
+        admin_ids.push(admin);
+    }
+
+    let mut configuration = default_configuration().await?;
+
+    configuration.no_direct_authentication = false;
+
+    configuration.trusted_identities = PreTrustedIdentities::Fixed(trusted_identities);
+
+    authority_node::start_node(ctx, &configuration).await?;
+
+    let mut admins = vec![];
+    for admin_id in admin_ids {
+        let sc = secure_channels
+            .create_secure_channel(ctx, &admin_id, route!["api"], SecureChannelOptions::new())
+            .await?;
+
+        let client = DirectAuthenticatorClient::new(
+            RpcClient::new(route![sc, DefaultAddress::DIRECT_AUTHENTICATOR], ctx).await?,
+        );
+
+        admins.push(Admin {
+            identifier: admin_id,
+            client,
+        });
+    }
+
+    Ok(admins)
+}
+
+#[ockam_macros::test]
+async fn test_only_admin_has_access(ctx: &mut Context) -> Result<()> {
+    use std::collections::HashMap;
+
+    let secure_channels = secure_channels();
+
+    let admins = setup(ctx, secure_channels.clone(), 1).await?;
+    let admin = &admins[0];
+
+    let member = secure_channels
+        .identities()
+        .identities_creation()
+        .create_identity()
+        .await?
+        .identifier()
+        .clone();
+
+    let mut attributes = HashMap::<&str, &str>::default();
+    attributes.insert("key", "value");
+
+    admin.client.add_member(member.clone(), attributes).await?;
+
+    let sc = secure_channels
+        .create_secure_channel(ctx, &member, route!["api"], SecureChannelOptions::new())
+        .await?;
+
+    let member_client = DirectAuthenticatorClient::new(
+        RpcClient::new(route![sc, DefaultAddress::DIRECT_AUTHENTICATOR], ctx).await?,
+    );
+
+    let timeout = Arc::new(AtomicBool::new(true));
+    let timeout_clone = timeout.clone();
+    ctx.runtime().spawn(async move {
+        let _ = member_client.list_member_ids().await;
+        timeout_clone.store(false, Ordering::Relaxed)
+    });
+    ctx.sleep(Duration::from_millis(50)).await;
+
+    assert!(timeout.load(Ordering::Relaxed));
+
+    ctx.stop().await?;
+
+    Ok(())
+}
+
+#[ockam_macros::test]
+async fn test_one_admin(ctx: &mut Context) -> Result<()> {
+    let secure_channels = secure_channels();
+
+    let admins = setup(ctx, secure_channels.clone(), 1).await?;
+    let admin = &admins[0];
+
+    let now = now()?;
+
+    let members = admin.client.list_member_ids().await?;
```

Comment:
I know this is obvious to you but it would be good to have comments indicating what is the intention here. Like "An admin is also a member of a given project", "the attributes of a admin do not expire", etc...

Reply:
Fixed